frame/contracts/src/exec.rs

```diff
@@ -349,6 +362,8 @@ enum FrameArgs<'a, T: Config, E> {
 		dest: T::AccountId,
 		/// If `None` the contract info needs to be reloaded from storage.
 		cached_info: Option<ContractInfo<T>>,
+		/// The executable whose `call` function is run.
+		executable: Option<E>,
```

Comment:
The name and comment is bad because it signals in no way that this is in connection to the delegate call:

```suggestion
		/// This frame was created by `seal_delegate_call` and hence uses a different code than what
		/// is stored at [`Self::dest`]. It also uses a different caller than the previous frames [`Self::dest`].
		delegated_call: (Option<E>, T::AccountId),
```

Also please don't use a tuple but create a struct with proper var names and docs for it. It is very important to keep this module documented well as it houses the main complexity.